### ISSUE TITLE
Calling toJS() (and other methods?) on Objects with _capacity: NaN results in infinite loop

### ISSUE THREAD

**Issue by batashi** [facebook/immutable-js#459]
Example:

```javascript
var list = Immutable.List([1, 2, 3, 4, 5]);
var newList = list.slice(0, NaN) // newList._capacity = NaN and newList.size = NaN
list.toJS() // causes infinite loop
```

I'm not sure whether the issue lies with the newList object being incorrectly formatted or toJS() not being able to parse that kind of object. It is probably the former as I believe I've gotten an infinite loop calling some method on that kind of object other than toJS().

**Comment by leebyron** [comment]
Thanks for the report!

**Comment by facebook-github-bot-4** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Comment by leebyron** [comment]
Fixed by #511
